### DISCUSSION TITLE
tiered storage

### DISCUSSION THREAD

**Discussion by eleaner** [moosefs/moosefs#579] [Q&A]
hi, I would like to reduce power consumption of my moosefs cluster

Would it make sense do create two classes - one FAST and one SLOW
FAST would be smaller, no spindown, SSD if I can afford it
SLOW would be spinning rust, with spindown but at least one HDD available - example labels U,D,D where U means always up and D means - will spin down.

then I create a script that would identify frequently accessed files (I did not figure it out yet in details) and move them to FAST while the others would go to SLOW

**Comment by chogata** [comment]
That's not gonna work, I'm afraid.
Ds, if their chunk server process is active, will never spin down. Remember, a chunk server is verifying chunks' integrity in a long, never ending loop, intervals between one check and the next are counted in seconds. No spin down possible.

If this is something that the community deems useful, we could add an option to disable chunk integrity check (setting per chunk server), with a big, **bold** (or ALL CAPS ;) ) warning, that this is not recommended. MooseFS is all about data safety, this check is part of that.

**Comment by eleaner** [comment]
Thank you.Disabling the integrity check does sound like a bad idea.However, what about setting up frequency of the check per storage class?Sent from my iPhoneOn 26 Jul 2024, at 10:57, Agata Kruszona-Zawadzka 'notifications at github.com' ***@***.***> wrote:﻿
That's not gonna work, I'm afraid.
Ds, if their chunk server process is active, will never spin down. Remember, a chunk server is verifying chunks' integrity in a long, never ending loop, intervals between one check and the next are counted in seconds. No spin down possible.
If this is something that the community deems useful, we could add an option to disable chunk integrity check (setting per chunk server), with a big, bold (or ALL CAPS ;) ) warning, that this is not recommended. MooseFS is all about data safety, this check is part of that.

**Comment by chogata** [comment]
A chunk server is not aware of storage classes - the master tells it to keep this and that chunk and the chunk server does as it's told. But the integrity check is an internal process of a chunk server that the master is not interested in (and not even aware of). So any tinkering with the integrity check can be done only for whole chunk servers. In your scenario, if you had your data in 3 copies, one copy on chunk server with label U and 2 copies on chunk servers with label D, you could switch off/lower frequency of the integrity check only on the servers with label D.

**Comment by eleaner** [comment]
This could work,Is there a parameter I could try and set?Sent from my iPhoneOn 26 Jul 2024, at 12:16, Agata Kruszona-Zawadzka 'notifications at github.com' ***@***.***> wrote:﻿
A chunk server is not aware of storage classes - the master tells it to keep this and that chunk and the chunk server does as it's told. But the integrity check is an internal process of a chunk server that the master is not interested in (and not even aware of). So any tinkering with the integrity check can be done only for whole chunk servers. In your scenario, if you had your data in 3 copies, one copy on chunk server with label U and 2 copies on chunk servers with label D, you could switch off/lower frequency of the integrity check only on the servers with label D.

**Comment by chogata** [comment]
I'm afraid there is no option right now, I said in my first comment it's something we could potentially add.

**Comment by eleaner** [comment]
I guess it makes sense to move the discussion to a different discussion topic focusing more on reducing energy consumption of moosefs cluster #580
